lxd/daemon.go

```diff
@@ -2109,6 +2109,45 @@ func (d *Daemon) nodeRefreshTask(heartbeatData *cluster.APIHeartbeat, isLeader b
 		}
 	}
 
+	// Check if a change to OVN chassis setup has occured.
+	hadOVNChassis := false
+	if d.lastNodeList != nil {
+		for _, n := range d.lastNodeList.Members {
+			for _, role := range n.Roles {
+				if role == db.ClusterRoleOVNChassis {
+					hadOVNChassis = true
+					break
+				}
+			}
+
+			if hadOVNChassis {
+				break
+			}
+		}
+	}
+
+	hasOVNChassis := false
+	for _, n := range heartbeatData.Members {
+		for _, role := range n.Roles {
+			if role == db.ClusterRoleOVNChassis {
+				hasOVNChassis = true
+				break
+			}
+		}
+
+		if hasOVNChassis {
+			break
+		}
+	}
+
+	if (!hadOVNChassis && hasOVNChassis) || (hadOVNChassis && !hasOVNChassis) {
+		// Detected a cluster-wide change in OVN setup requiring a restart of all OVN networks.
+		err := networkRestartOVN(d.State())
+		if err != nil {
+			logger.Error("Error restarting OVN networks", log.Ctx{"err": err})
```

Comment:
This should set `stateChangeTaskFailure=true` here so its re-tried on next heartbeat.